src/pointer-analysis/value_set_dereference.h

```diff
@@ -74,6 +74,13 @@ class value_set_dereferencet final
     irep_idt language_mode,
     const namespacet &ns);
 
+  static optionalt<exprt> build_reference_to_value_set_element(
```

Comment:
Please document.

Reply:
What I want to call it is `build_reference_to`, but that's already taken by the function it's wrapping (which doesn't actually build a reference).

The reason I have a wrapper function is that `build_reference_to` makes it easy to get objects, but not pointers, e.g. it gives you `global`, not `&global`, and for null pointers it uses the special value `nil_exprt`, and doesn't give me the `null_pointer_exprt` that I want. That makes sense because I think it's only used for `dereference_rec`, which only cares about dereferenced pointers.  Now that I've read through that function a few more times, I think a better solution would be to add a field to `value_set_dereferencet::valuet` to put `&global` and `null_pointer_exprt` into. What would you say to that?

Reply:
I have done this now